prototype_source/vulkan_workflow.rst

```diff
@@ -0,0 +1,165 @@
+
+Building PyTorch with Vulkan backend
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+PyTorch 1.7 does not include Vulkan backend by default. The main switch to include Vulkan backend is cmake option ``USE_VULKAN``, that can be set by environment variable ``USE_VULKAN``.
+
+Vulkan wrapper
+^^^^^^^^^^^^^
+
+By defaulti, Vulkan library will be loaded at runtime using the vulkan_wrapper library. If you specify the environment variable ``USE_VULKAN_WRAPPER=0`` libvulkan will be linked directly.
+
+Desktop build
+^^^^^^^^^^^^^
+
+Download VulkanSDK from https://vulkan.lunarg.com/sdk/home and set environment variable ``VULKAN_SDK``
+
+::
+
+    cd PYTORCH_ROOT
+    USE_VULKAN=1 USE_VULKAN_WRAPPER=0  python setup.py install
+
+Android build
+^^^^^^^^^^^^^
+::
+
+    cd PYTORCH_ROOT
+    ANDROID_ABI=arm64-v8a USE_VULKAN=1 sh ./scripts/build_android.sh
+
+
+Model preparation
+-----------------
+
+Install torch and torchvision, get the default pretrained float model.
+
+::
+
+    pip install torch torchvision
+
+Python script to save pretrained mobilenet_v2 to a file:
+
+::
+
+    import torch
+    import torchvision
+
+    model = torchvision.models.mobilenet_v2(pretrained=True)
+    model.eval()
+    script_model = torch.jit.script(model)
+    torch.jit.save(script_model, “mobilenet2.pt”)
+
+PyTorch 1.7 Vulkan backend supports only float 32bit operators. The default model needs additional step that will optimize operators fusing 
+
+::
+
+    from torch.utils.mobile_optimizer import optimize_for_mobile
+    script_model_vulkan = optimize_for_mobile(script_model, backend=’Vulkan’)
+    torch.jit.save(script_model_vulkan, "mobilenet2-vulkan.pt")
+
+The result model can be used only on Vulkan backend as it contains specific to the Vulkan backend operators.
+
+Using Vulkan backend in code
+----------------------------
+
+C++ API
+-------
+
+::
+
+    at::is_vulkan_available()
+    auto tensor = at::rand({1, 2, 2, 3}, at::device(at::kCPU).dtype(at::kFloat));
+    auto tensor_vulkan = t.vulkan();
+    auto module = torch::jit::load(“$PATH”);
+    auto tensor_output_vulkan = module.forward(inputs)
+    auto tensor_output = tensor_output.cpu()
+
+``at::is_vulkan_available()`` function tries to initialize Vulkan backend and if Vulkan device is successfully found and context is created - it will return true, false otherwise.
+
+``.vulkan()`` function called on Tensor will copy tensor to Vulkan device, and for operators called with this tensor as input - the operator will run on Vulkan device, and its output will be on the Vulkan device.
+
+``.cpu()`` function called on Vulkan tensor will copy its data to CPU tensor (default)
+
+Operators called with a tensor on a Vulkan device as an input will be executed on a Vulkan device. If an operator is not supported for the Vulkan backend the exception will be thrown.
+List of supported operators:
```

Comment:
Nit: Add a  space above line 90 for readability; otherwise, it's rendering on the same line as the previous sentence.